sycl/include/sycl/ext/oneapi/experimental/group_helpers_sorters.hpp

```diff
@@ -124,6 +302,216 @@ template <typename T> struct ConvertToComp<T, sorting_order::descending> {
+          sorting_order Order = sorting_order::ascending,
+          size_t ElementsPerWorkItem = 1, unsigned int BitsPerPass = 4>
+class group_key_value_sorter {
+
+  std::byte *scratch;
+  uint32_t first_bit;
+  uint32_t last_bit;
+  size_t scratch_size;
+
+  static constexpr uint32_t bits = BitsPerPass;
+
+public:
+  template <std::size_t Extent>
+  group_key_value_sorter(
+      sycl::span<std::byte, Extent> scratch_,
+      const std::bitset<sizeof(T) * CHAR_BIT> mask =
+          std::bitset<sizeof(T) * CHAR_BIT>(
+              (std::numeric_limits<unsigned long long>::max)()))
+      : scratch(scratch_.data()), scratch_size(scratch_.size()) {
+    static_assert(
+        (std::is_arithmetic<T>::value || std::is_same<T, sycl::half>::value),
+        "radix sort is not usable");
+    first_bit = 0;
+    while (first_bit < mask.size() && !mask[first_bit])
+      ++first_bit;
+
+    last_bit = first_bit;
+    while (last_bit < mask.size() && mask[last_bit])
+      ++last_bit;
+  }
+
+  // template <typename Group> std::tuple<T, U> operator()(Group g, T key, U
+  // val) {
+  template <typename Group> std::tuple<T, U> operator()(T key, U val) {
+    static_assert(ElementsPerWorkItem == 1, "ElementsPerWorkItem must be 1");
+    T key_result[]{key};
+    U val_result[]{val};
+#ifdef __SYCL_DEVICE_ONLY__
+
+#if 0
+    sycl::detail::privateStaticSort<
+        /*is_key_value=*/true,
+        /*is_blocked=*/true, 1, bits>(
+        g, key_result, val_result, Order == sorting_order::ascending,
+        scratch, sycl::detail::Builder::getNDItem<Group::dimensions>(),
+        first_bit, last_bit);
+#endif
+#endif
+    key = key_result[0];
+    val = val_result[0];
+    return {key, val};
+  }
+
+  template <typename Group, typename Properties>
+  void operator()(Group g, sycl::span<T, ElementsPerWorkItem> keys,
+                  sycl::span<U, ElementsPerWorkItem> vals,
+                  Properties property) {
+    (void)g;
+    (void)keys;
+    (void)vals;
+    (void)property;
+#ifdef __SYCL_DEVICE_ONLY__
+    sycl::detail::privateStaticSort<
+        /*is_key_value=*/true, std::is_same_v<Properties, detail::is_blocked>,
+        Order == sorting_order::ascending, ElementsPerWorkItem, bits>(
+        g, keys.data(), vals.data(), scratch, first_bit, last_bit);
+#endif
+  }
+
+  static constexpr std::size_t memory_required(sycl::memory_scope scope,
+                                               std::size_t range_size) {
+    (void)scope;
+    return (std::max)(range_size * ElementsPerWorkItem *
+                              (sizeof(T) + sizeof(U)) +
+                          2 * alignof(uint32_t),
+                      range_size * (1 << bits) * sizeof(uint32_t) + alignof(T));
+  }
+};
+
+} // namespace radix_sorters
+
+enum class group_algorithm_data_placement { blocked, striped };
+
 template <typename ValT, sorting_order OrderT = sorting_order::ascending,
           unsigned int BitsPerPass = 4>
 class radix_sorter {
```

Comment:
We can remove this class since it's replace by `radix_sorters`

Reply:
Removed